src/OSPSuite.Core/Domain/Services/ParameterIdentifications/ParameterIdentificationRunner.cs

```diff
@@ -39,6 +39,9 @@ public class ParameterIdentificationRunner : IParameterIdentificationRunner
 
       public async Task Run(ParameterIdentification parameterIdentification)
       {
+         if(parameterIdentification.IsLoaded != true) 
```

Comment:
Why the check here? I would assume that Load should make sure it's not loaded already. CAn we double check this? TELL don't ASK principle

Reply:
and also add a test please.

@benjaperez1983 In short, when we fix a bug, we try to always write a test. If it's not UI related, we ALWAYS write a test

Reply:
This is because we have a context menu that allows the PI to run without opening

Reply:
ah, ok, here's the long story - This issue was filed as 'can't open', but what probably happened is we have two bugs here. First - when user double clicked on the PI, the top context menu was executed. Recently that was mistakenly changed to 'Run'. So while Pavel thought he was opening the PI, it was actually running. We had fixed that bug in Core. However, we noticed that if we ran the PI without opening it, the bug still happens.

I'll change the PR and issue descriptions

Reply:
I thought you're asking why are we doing loading during the run, but you're not asking that.

You're asking if the `_executionContext.Load` already checks the `IsLoaded`, if so, we don't have to also check it here.